diff --git a/bonsai.sh b/bonsai.sh
index cee904d..fb234d1 100755
--- a/bonsai.sh
+++ b/bonsai.sh
@@ -379,7 +379,7 @@ init() {
 
 	# setup temp file for caching times of each growth
 	mkdir -p /tmp/bonsai.sh
-	tmpFile="$(mktemp -p /tmp/bonsai.sh bonsai.sh.XXXXXXXX)"
+	tmpFile="$(gmktemp -p /tmp/bonsai.sh bonsai.sh.XXXXXXXX)"
 }
 
 grow() {
@@ -592,7 +592,7 @@ branch() {
 		# print what we have so far
 		if ((live)); then
 			( time -p display ) 2>"$tmpFile"
-			elapsed="$(head "$tmpFile" -n 1 | awk '{print $2}' )"
+			elapsed="$(ghead "$tmpFile" -n 1 | awk '{print $2}' )"
 			# if this step took less than $stepTime, sleep until $stepTime is met
 			timeLeft="$(printf '%s\n' "$timeStep - $elapsed" | bc -l)"
 			[ "$(printf '%s\n' "($timeLeft) > 0" | bc -l)" -eq 1 ] && sleep "$timeLeft"
